apps/server/src/shared/domain/entity/task.entity.ts

```diff
@@ -59,8 +64,8 @@ export class Task extends BaseEntityWithTimestamps implements ILearnroomElement,
 	publicSubmissions?: boolean;
 
 	@Index()
-	@ManyToOne('User', { fieldName: 'teacherId', nullable: true })
-	creator?: User;
+	@ManyToOne('User', { fieldName: 'teacherId' })
```

Comment:
I think in the past, students could also create tasks (which are private) - so we might have tasks in db with no teacherId

Reply:
I guess the naming "teacher" is misleading here. The correct name of this field would be creatorId, because in the case of a student being the creator, thies field would contain a student id.

Reply:
Yes this is the creator and it is used in legacy code as creator, but wrong named in collection. (all role can be added)